### PR TITLE
Cookbook: Background processing, Orientation, and Updates

### PR DESCRIPTION
**Pull request by brianegan**
Hey hey :) Another few recipes for the cookbook, based on suggestions from the voting thread.

  * Parsing JSON in a background isolate
  * Building a UI depending on the orientation
  * Update the Key-Value example to improve grammar / layout

### CODE REVIEW

File: cookbook/design/orientation.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+---
+layout: page
+title: "Updating the UI based on orientation"
+permalink: /cookbook/design/orientation/
+---
+
+In certain cases, it can be handy to update the design of an app when the user
```
Comment (Hixie) [inline cookbook/design/orientation.md]: you could just use OrientationBuilder for this

File: cookbook/networking/background-parsing.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+## 4. Move this work to a separate isolate
+
+If you run the `fetchPhotos` function on a slower phone, you may notice the app 
+freezes for a brief moment as it parses and converts the json. This is jank, 
+and we want to be rid of it!
+
+So how can we do that? By moving the parsing and conversion to a background
+isolate. To do so, we'll need to:
+
+  1. Create an `IsolateRunner`. This will allow us to run the `fetchPhotos`
```
Comment (Hixie) [inline cookbook/networking/background-parsing.md]: you may wish to consider the compute() method for this

Comment (brianegan) [reply cookbook/networking/background-parsing.md]: Ah, perfect, that's even simpler! Thanks :)

File: cookbook/design/orientation.md (hunk 1 above)
Comment (brianegan) [reply cookbook/design/orientation.md]: Sounds good. I wasn't sure if that was always the right tool for the job, but happy to go that route if it's more idiomatic.

**Comment by brianegan** [comment]
Thanks for the feedback! It's getting late here, but I'll make these changes soon :)

File: cookbook/networking/background-parsing.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+```yaml
+dependencies:
+  http: <latest_version>
+  isolate: <latest_version>
+```
+  
+## 2. Make a network request
+
+In this example, we'll fetch a json large document that contains a list of 5000
```
Comment (xqwzts) [inline cookbook/networking/background-parsing.md]: "a large JSON document"?

File: cookbook/design/orientation.md (hunk 1 above)
Comment (brianegan) [reply cookbook/design/orientation.md]: Fixed.

**Comment by brianegan** [comment]
Changes applied, ready for review.

**Comment by brianegan** [comment]
@mjohnsullivan Any chance you could take a look at this one and comment / merge?

**Comment by Hixie** [comment]
I sent you an invite to add you to the repo. Please read https://flutter.io/design-principles/ for the rules of how to operate in this repo. Then you can merge your own patches. :-)

LGTM

**Comment by brianegan** [comment]
@Hixie Thanks so much for taking a look and providing me with access to merge! I'll be sure to follow the guidelines described in that doc.

**Pull request merged**